Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/HttpUtility.cs

```diff
@@ -0,0 +1,861 @@
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Globalization;
+using System.IO;
+using System.Security.Permissions;
+using System.Text;
+using System.Web.Util;
+
+namespace System.Web
+{
+
+#if !MOBILE
+    // CAS - no InheritanceDemand here as the class is sealed
+    [AspNetHostingPermission(SecurityAction.LinkDemand, Level = AspNetHostingPermissionLevel.Minimal)]
+#endif
+    public sealed class HttpUtility
+    {
+        sealed class HttpQSCollection : NameValueCollection
+        {
+            public override string ToString()
+            {
+                int count = Count;
+                if (count == 0)
+                {
+                    return "";
+                }
+
+                StringBuilder sb = new StringBuilder();
+                string[] keys = AllKeys;
+
+                for (int i = 0; i < count; i++)
+                {
+                    sb.AppendFormat("{0}={1}&", keys[i], UrlEncode(this[keys[i]]));
+                }
+
+                if (sb.Length > 0)
+                {
+                    sb.Length--;
+                }
+
+                return sb.ToString();
+            }
+        }
+
+        #region Methods
+
+        public static void HtmlAttributeEncode(string s, TextWriter output)
+        {
+            if (output == null)
+            {
+                throw new ArgumentNullException(nameof(output));
+            }
+
+            HttpEncoder.Current.HtmlAttributeEncode(s, output);
+        }
+
+        public static string HtmlAttributeEncode(string s)
+        {
+            if (s == null)
+            {
+                return null;
+            }
+
+            using (var sw = new StringWriter())
+            {
+                HttpEncoder.Current.HtmlAttributeEncode(s, sw);
+                return sw.ToString();
+            }
+        }
+
+        public static string UrlDecode(string str)
+        {
+            return UrlDecode(str, Encoding.UTF8);
+        }
+
+        static char[] GetChars(MemoryStream b, Encoding e)
+        {
+            return e.GetChars(b.GetBuffer(), 0, (int)b.Length);
+        }
+
+        static void WriteCharBytes(IList buf, char ch, Encoding e)
+        {
+            if (ch > 255)
+            {
+                foreach (byte b in e.GetBytes(new char[] { ch }))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)